drivers/gnss/gnss_ubx_common.c

```diff
@@ -27,19 +27,24 @@ void gnss_ubx_common_pvt_callback(struct modem_ubx *ubx, const struct ubx_frame
 	if ((nav_pvt->flags & UBX_NAV_PVT_FLAGS_GNSS_FIX_OK) &&
 	    !(nav_pvt->nav.flags3 & UBX_NAV_PVT_FLAGS3_INVALID_LLH)) {
 
-		switch (nav_pvt->fix_type) {
-		case UBX_NAV_FIX_TYPE_DR:
-		case UBX_NAV_FIX_TYPE_GNSS_DR_COMBINED:
+		if (nav_pvt->flags & UBX_NAV_PVT_FLAGS_GNSS_CARR_SOLN_FLOATING) {
```

Comment:
MISRA rule 15.7 -- All if else if constructs shall be terminated with an else statement